rotkehlchen/globaldb/upgrades/manager.py

```diff
@@ -171,3 +192,42 @@ def _perform_single_upgrade(
             'INSERT OR REPLACE INTO settings(name, value) VALUES(?, ?)',
             ('version', str(to_version)),
         )
+
+
+def configure_globaldb(
+        global_dir: Path,
+        db_filename: str,
+        connection: 'DBConnection',
+        msg_aggregator: 'MessagesAggregator',
+        globaldb: 'GlobalDBHandler | None' = None,
+) -> None:
+    """Configure the global database and handle schema upgrades.
+
+    - global_dir: Directory containing the global database
+    - db_filename: Name of the database file (typically global.db)
+    - connection: The database connection object
+    - msg_aggregator: Message aggregator for logging
+    - globaldb: Optional handler instance - determines whether asset updates are attempted
+
+    May raise:
+        - DBSchemaError if the database schema is invalid.
+    """
+    is_fresh_db = maybe_upgrade_globaldb(
+        globaldb=globaldb,
+        connection=connection,
+        global_dir=global_dir,
+        db_filename=db_filename,
+        msg_aggregator=msg_aggregator,
+    )
+    connection.executescript('PRAGMA foreign_keys=on;')
```

Comment:
nah nah. i didn't set it now -- i was just splitting the implementation of the funtion below into `initialize_globaldb()` and `configure_globaldb()` https://github.com/rotki/rotki/blob/67b1dbfc379ecd0bd9a7d348ea3dedbd9c6528d6/rotkehlchen/globaldb/handler.py#L145-L183

Reply:
Why is this here? The fact that it was here before does not answer much. Is it something that always needs to be specified at initialization?

Keep in mind it's also done in `DB_SCRIPT_CREATE_TABLES` if it's a `fresh_db`

Reply:
>  Is it something that always needs to be specified at initialization?

yeah since its not a fresh database and foreign keys are not turned on by default when a db is created.